sycl/doc/design/OffloadDesign.md

```diff
@@ -0,0 +1,326 @@
+provided to the wrapper to give more control over what device libraries are
+pulled in during the device link.
+
+### Ahead Of Time Compilation
+
+The updated offloading model will integrate the Ahead of Time (AOT) compilation
+behaviors into the clang-linker-wrapper.  The actions will typically take place
+after the device link and post link steps.
+
+Regardless of the AOT target, the flow is similar, only modifying the offline
+compiler that is used to create the target device image.  It is expected that
+the offline compiler will also use unique command lines specific to the tool
+to create the image.
+
+To support the needed option passing triggered by use of the
+`-Xsycl-target-backend` option and implied options based on the optional
+device behaviors for AOT compilations for GPU new command line interfaces
+are needed to pass along this information.
+
+| Target | Triple        | Offline Tool   | Option for Additional Args |
+|--------|---------------|----------------|----------------------------|
+| CPU    | spir64_x86_64 | opencl-aot     | `--cpu-tool-arg=<arg>`     |
+| GPU    | spir64_gen    | ocloc          | `--gen-tool-arg=<arg>`     |
+| FPGA   | spir64_fpga   | aoc/opencl-aot | `--fpga-tool-arg=<arg>`    |
+
+*Table: Ahead of Time Info*
+
+To complete the support needed for the various targets using the
+`clang-linker-wrapper` as the main interface, a few additional options will
+be needed to communicate from the driver to the tool.  Further details of usage
+are given further below.
+
+| Option Name                  | Purpose                                      |
+|------------------------------|----------------------------------------------|
+| `--fpga-tool-deps=<arg>`     | Comma separated list of dependency files used for FPGA hardware compiles using `aoc` |
+| `--parallel-link-sycl=<arg>` | Provide the number of parallel jobs that will be used when processing with `llvm-foreach` |
+| `--no-sycl-device-lib=<arg>` | Provide the list of device libraries to restrict from linking during device link |
+
+*Table: Additional Options for clang-linker-wrapper*
+
+The `clang-linker-wrapper` provides an existing option named `-wrapper-jobs`
+that may be useful for our usage instead of creating a new option specific
+to `llvm-foreach` processing.
+
+#### spir64_gen support
+
+Compilation behaviors involving AOT for GPU involve an additional call to
+the OpenCL Offline compiler (OCLOC).  This call occurs after the post-link
+step performed by sycl-post-link.  Additional options passed by the user
+via the `-Xsycl-target-backend=spir64_gen <opts>` command as well as the
+implied options set via target options such as `-fsycl-targets=intel_gpu_skl`
+will be processed by a new options to the wrapper, `--gen-tool-arg=<arg>`
+
+#### spir64_fpga support
+
+Compilation behaviors involving AOT for FPGA involve an additional call to
+the either `aoc` (for Hardware) or `opencl-aot` (for Simulation).  This call
```

Comment:
`opencl-aot` is only needed for what we call emulation (x86 emulation of the user kernel). `aoc` is used for both hardware and simulation (cycle-accurate simulation of the generated Verilog code) platforms.

Reply:
Thanks - I will correct this.